pkg/crc/constants/constants.go

```diff
@@ -115,6 +116,19 @@ func GetDefaultBundle(preset crcpreset.Preset) string {
 	return bundles[runtime.GOOS]
 }
 
+func GetBundleNameFromBundlePathString(preset crcpreset.Preset, bundlePath string) string {
+	switch {
+	case strings.HasPrefix(bundlePath, "http://"), strings.HasPrefix(bundlePath, "https://"):
+		return filepath.Base(bundlePath)
+	case strings.HasPrefix(bundlePath, "docker://"):
+		tag := strings.Split(bundlePath, ":")[2]
+		bundles := bundleForOs(preset, tag)
```

Comment:
> If we are requiring the tag to match what we use on quay, then I'd prefer that we also require for now that the repo name matches the ones we use on quay

@cfergeau  you mean the org name `crcont` or only the container image name `ocp-bundle`/`okd-bundle`/`podman-bundle` ? Also even using those how do we able to build the bundle name 
`ocp-bundle` => bundle name should be `crc_<hypervisor>.<tag>.crcbundle` ?

Shouldn't current method is better in case user choose the preset `openshift` but chose image of `podman` then in current scenario we fail in case there is no podman image or pull the correct one but if we don't consider the `preset` then we are going to pull wrong image for wrong preset and error out after extraction.

Reply:
> Shouldn't current method is better in case user choose the preset openshift but chose image of podman then in current scenario we fail in case there is no podman image or pull the correct one but if we don't consider the preset then we are going to pull wrong image for wrong preset and error out after extraction.

What bothers me is that the behaviour is totally different from the behaviour of the other 2 methods (https and file). I'd prefer to have something consistent for all 3 methods, bundle filename comes from the URL, and nothing else.

Reply:
> What bothers me is that the behaviour is totally different from the behaviour of the other 2 methods (https and file). I'd prefer to have something consistent for all 3 methods, bundle filename comes from the URL, and nothing else.

I don't know how we can do with quay, in that case we explicitly need to ask user to have the image name as bundle filename on the mirror. Or we can use preset for https and file also for consistency to make sure user use the correct bundle of preset as part of validation. This function still be inconsistent but validation side we able to validate `https/file` method also depend on preset.

```
func validBundleForPreset(inputString string, preset crcpreset.Preset) error {
	if strings.HasPrefix(inputString, "docker://") || preset == crcpreset.OpenShift {
		return nil
	}
	if !strings.Contains(filepath.Base(inputString), preset.String()) {
		return fmt.Errorf("invalid %s bundle for %s preset", filepath.Base(inputString), preset)
	}
	return nil
}
```

Reply:
Could be something like:
```
switch repoName {
    case "podman-bundle: preset = preset.Podman
    case "ocp-bundle": preset = preset.OpenShift
    ...
}
```
We already put requirements on the version tag, so we could also do this on repo name. This would then be made more flexible at a later point.

Reply:
@cfergeau Just for sake of consistency which is not even correct because as par current behavior user can choose different bundle with different preset and only get to know this metadata (after bundle extraction) I would better use preset validation for even http/file and tell that early on to user then wait for extraction.